### PR TITLE
Clippy suggestion cleanup.

### PR DESCRIPTION
**Pull request by 0ndorio**
I spent some time to fix most of the clippy suggestions that do not interfere with the function signatures or internal code structuring. As this is neither an API change nor a new feature I skipped the update of `docs/CHANGELOG.md`.

Feel free to nitpick If this includes any unwanted changes or if I missed something important. I will try to react as fast as possible.

I would also suggest to review the two commits independently as the second one only contains automatically applied changes made by rustfmt (`rustc 1.29.2 (17a9dc751 2018-10-05)
`, `rustfmt 0.99.1-stable (da17b68 2018-08-04)`).

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -422,6 +429,7 @@ impl<S, E, X> ApplicationBuilder<S, E, X> {
     /// game.run();
     /// ~~~
 
+    // TODO: Unused lifetime. Is it required?
```
Comment (Xaeroxe) [inline src/app.rs]: I'd venture a "no"

Comment (Rhuagh) [reply src/app.rs]: No, it's a remnant of old code that's been refactored atleast three times :)

File: amethyst_animation/src/systems/control.rs [hunk 2]
```diff
@@ -439,13 +439,15 @@ where
...
+    let still_loading = animation
         .nodes
         .iter()
         .any(|&(ref node_index, _, ref sampler_handle)| {
             !hierarchy.nodes.contains_key(node_index)
                 || sampler_storage.get(sampler_handle).is_none()
-        }) {
+        });
+
+    if still_loading {
```
Comment (Rhuagh) [inline amethyst_animation/src/systems/control.rs]: What's the motivation for this change?

Comment (0ndorio) [reply amethyst_animation/src/systems/control.rs]: The idea was to take the "complex" if statement and introduce an intermediate variable name that represents the intention of the statement. I assume that the comment above the statement could be seen as doing the same as the its only the check for a precondition.

File: amethyst_renderer/src/pipe/pass.rs [hunk 3]
```diff
@@ -16,8 +16,10 @@ pub trait Pass: for<'a> PassData<'a> {
     /// The pass is given an opportunity to compile shaders and store them in an `Effect`
     /// which is then passed to the pass in `apply`.
     fn compile(&mut self, effect: NewEffect) -> Result<Effect>;
+
     /// Called whenever the renderer is ready to apply the pass.  Feed commands into the
     /// encoder here.
+    // TODO: Not one of the implementations uses lifetime 'b. Is it required?
```
Comment (0ndorio) [inline amethyst_renderer/src/pipe/pass.rs]: What about this one?

File: amethyst_animation/src/resources.rs [hunk 4]
```diff
@@ -134,7 +134,7 @@ where
 /// Defines the hierarchy of nodes that a single animation can control.
 /// Attached to the root entity that an animation can be defined for.
 /// Only required for animations which target more than a single node or entity.
-#[derive(Debug, Clone)]
+#[derive(Debug, Default, Clone)]
```
Comment (torkleyy) [inline amethyst_animation/src/resources.rs]: * should use the default in `new`
* should use `derivative` as `derive` only implements it for `T: Default`

Comment (0ndorio) [reply amethyst_animation/src/resources.rs]: thx! totally missed that.

File: amethyst_animation/src/systems/control.rs (hunk 2 above)
Comment (0ndorio) [reply amethyst_animation/src/systems/control.rs]: Restored the original version.

File: amethyst_renderer/src/pipe/pass.rs (hunk 3 above)
Comment (Rhuagh) [reply amethyst_renderer/src/pipe/pass.rs]: This one is needed, because the pass data does not have the same lifetime as the pass.

Comment (0ndorio) [reply amethyst_renderer/src/pipe/pass.rs]: Okay. Was just curious because all implementations of `Pass` I could discover either utilized `'a` a second time (e.g. `Pass for DrawSprite`) or the anonymous lifetime  (e.g. `Pass for DrawUi`) but never `'b` for `PassData`.

**Comment by 0ndorio** [comment]
Should I go an rebase on current master as the branch was updated in the mean time?

File: amethyst_renderer/src/pipe/effect/mod.rs [hunk 5]
```diff
@@ -94,8 +90,11 @@ impl Effect {
         match self.const_bufs.get(name.as_ref()) {
             Some(i) => {
                 let raw = &self.data.const_bufs[*i];
-                enc.update_buffer::<T>(unsafe { mem::transmute(raw) }, &data[..], 0)
-                    .expect("Failed to update buffer (TODO: replace expect)");
+                enc.update_buffer::<T>(
+                    unsafe { &*(raw as *const RawBuffer<_> as *const Buffer<_, _>) },
```
Comment (jojolepro) [inline amethyst_renderer/src/pipe/effect/mod.rs]: I can't read this, so I'd like to know if this was reviewed by the other reviewers.

Comment (torkleyy) [reply amethyst_renderer/src/pipe/effect/mod.rs]: Yes, it was ;)

**Comment by magnonellie** [comment]
@0ndorio Sorry for pinging you like this, but I couldn't find you on Discord :sweat_smile: 

How did you get clippy to run on Amethyst? For me it's just throwing errors on dependencies using the rustup install method. I would like to integrate clippy into our CI at least in some capacity so being able to get it running locally is a pretty good first step I think. :stuck_out_tongue:
